test/pkg/environment/common/environment.go

```diff
@@ -144,42 +164,45 @@ func NewClient(ctx context.Context, config *rest.Config) client.Client {
 	return c
 }
 
-func (env *Environment) DefaultNodeClass() *v1alpha1.KWOKNodeClass {
-	return &v1alpha1.KWOKNodeClass{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: test.RandomName(),
-		},
+func (env *Environment) DefaultNodePool(nodeClass client.Object) *v1.NodePool {
+	nodePool := &v1.NodePool{}
+	if lo.FromPtr(nodePoolPath) == "" {
+		nodePool = object.Unmarshal[v1.NodePool](defaultNodePool)
+	} else {
+		file := lo.Must1(os.ReadFile(lo.FromPtr(nodePoolPath)))
+		lo.Must0(yaml.Unmarshal(file, nodePool))
 	}
-}
 
-func (env *Environment) DefaultNodePool(nodeClass *v1alpha1.KWOKNodeClass) *v1.NodePool {
-	nodePool := test.NodePool()
 	nodePool.Spec.Template.Spec.NodeClassRef = &v1.NodeClassReference{
-		Name:  nodeClass.Name,
-		Kind:  object.GVK(nodeClass).Kind,
-		Group: object.GVK(nodeClass).Group,
+		Kind:  env.DefaultNodeClass.GroupVersionKind.Kind,
+		Group: env.DefaultNodeClass.GroupVersionKind.Group,
+		Name:  env.DefaultNodeClass.GetName(),
 	}
-	nodePool.Spec.Template.Spec.Requirements = []v1.NodeSelectorRequirementWithMinValues{
-		{
-			NodeSelectorRequirement: corev1.NodeSelectorRequirement{
-				Key:      corev1.LabelOSStable,
-				Operator: corev1.NodeSelectorOpIn,
-				Values:   []string{string(corev1.Linux)},
-			},
-		},
-		{
-			NodeSelectorRequirement: corev1.NodeSelectorRequirement{
-				Key:      v1.CapacityTypeLabelKey,
-				Operator: corev1.NodeSelectorOpIn,
-				Values:   []string{v1.CapacityTypeOnDemand},
-			},
-		},
-	}
-	nodePool.Spec.Disruption.ConsolidateAfter = v1.MustParseNillableDuration("Never")
-	nodePool.Spec.Template.Spec.ExpireAfter.Duration = nil
-	nodePool.Spec.Limits = v1.Limits(corev1.ResourceList{
-		corev1.ResourceCPU:    resource.MustParse("1000"),
-		corev1.ResourceMemory: resource.MustParse("1000Gi"),
-	})
+
 	return nodePool
 }
+
+func decodeNodeClass() *GenericNodeClass {
+	// Open the file
+	if lo.FromPtr(nodeClassPath) == "" {
+		return &GenericNodeClass{
+			Unstructured:     lo.FromPtr(object.Unmarshal[unstructured.Unstructured](defaultNodeClass)),
+			GroupVersionKind: object.GVK(&v1alpha1.KWOKNodeClass{}),
+		}
+	}
+
+	file := lo.Must1(os.Open(lo.FromPtr(nodeClassPath)))
+	content := lo.Must1(io.ReadAll(file))
+
+	decoder := serializeryaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	u := &unstructured.Unstructured{}
+	ul := &unstructured.UnstructuredList{}
+	_, gvk, _ := decoder.Decode(content, nil, u)
+	u.SetGroupVersionKind(lo.FromPtr(gvk))
+	ul.SetGroupVersionKind(lo.FromPtr(gvk))
+
+	return &GenericNodeClass{
+		Unstructured:     lo.FromPtr(u),
+		GroupVersionKind: lo.FromPtr(gvk),
```

Comment:
Why do you need to create a separate struct for this? Is this just an `unstructured.Unstructured` or is there something that you are trying to do in the future to extend additional methods off of this object?

Reply:
Yeah, I was think we could expanded this struct to contain more configuration information, but I think we can wait on that when the requirement arises